versionedRouter.js

```diff
@@ -90,6 +93,38 @@ const functionsEnabled = () => {
   return areFunctionsEnabled === 1;
 };
 
+async function handleCdkV2(destName, parsedEvent) {
+  try {
+    const workflowEngine = getWorkflowEngine(destName);
+
+    const result = await workflowEngine.execute(parsedEvent);
+    // TODO: Handle remaining output scenarios
+    return result.output;
+  } catch (err) {
+    const errorInfo = getErrorInfo(err);
+
+    // TODO: Bump the error priority even further as it's an unhandled error in the CDK
+    const errObj = generateErrorObject(
+      errorInfo,
+      destName,
+      TRANSFORMER_METRIC.TRANSFORMER_STAGE.TRANSFORM
+    );
+
+    // Dump the raw error
+    logger.error(err);
+
+    return {
+      metadata: parsedEvent.metadata,
+      statusCode: errObj.status,
+      error: errObj.message || "Error occurred while processing the payload",
+      statTags: {
+        errorAt: TRANSFORMER_METRIC.ERROR_AT.PROC,
```

Comment:
It is not always going to be a `PROC` only as in CDK V2, we  inherently support have support for router transformation type of logic as well.
May be we need to accept one more argument like `errorAt` and pass it explicitly from where this method is called.

Something like below should solve our problem. If there is a more intuitive solution in mind, even that works
```javascript
async function handleCdkV2(destName, parsedEvent, errorAt) {
...
    statTags: {
        errorAt,
        ...
    }
}

// handleDest - processor transformation
    respEvents = await handleCdkV2(destination, parsedEvent, TRANSFORMER_METRIC.ERROR_AT.PROC);

// routerHandleDest - router transformation
   respEvents = await handleCdkV2(destination, parsedEvent, , TRANSFORMER_METRIC.ERROR_AT.RT);
```

Reply:
I added it as a default value which can get overridden if the stat tags (that user sent) contain it.
I've incorporated your suggestion.

Reply:
Rather than having a default we should thrown an error if `flowType` is not defined, as this is going to be at `versionedRouter` level we should be good with even throwing an error.

Reply:
Throw an error and fail the event?

At the end of the day, it's just another meta information in the metrics. If we realize any bug after the fact, we can always go and fix it.
Metrics is an auxiliary functionality. It shouldn't interfere with real processing any day.

Reply:
It's not going to interfere the real processing actually. How does that happen is ?

We are going to call this handler for CDK V2 at versionedRouter level only and since while adding this handler elsewhere and testing it, an error will be thrown while dev testing only, which is when the fix also would happen. Any additional protection can be provided by adding unit-test cases

It also makes sense to may be to not throw an error but we don't have a context of the caller to know at which handler did the error actually occur. Without that context, it would be very difficult to debug. So instead have a default `errorAt` tag known as `UNKNOWN` may be to indicate that the error occurred from an unknown caller. This is more apt than having `PROC` as the default for `errorAt` tag

Reply:
`errorAt` currently serves as a pointer to the endpoint(more like handler of that endpoint) at which the error has occurred. I'm just making sure of the definition and how it is being used for alerting

Reply:
Added `UNKNOWN` type.